src/dune_rules/preprocessing.ml

```diff
@@ -444,15 +442,16 @@ let ppx_driver_and_flags_internal sctx ~loc ~expander ~lib_name ~flags libs =
   (ppx_driver_exe sctx libs, flags @ cookies)
 
 let ppx_driver_and_flags sctx ~lib_name ~expander ~scope ~loc ~flags pps =
-  let open Result.O in
-  let* libs = Lib.DB.resolve_pps (Scope.libs scope) pps in
-  let* exe, flags =
-    ppx_driver_and_flags_internal sctx ~loc ~expander ~lib_name ~flags libs
-  in
-  let+ driver =
-    Lib.closure libs ~linking:true >>= Driver.select ~loc:(User_file (loc, pps))
-  in
-  (exe, driver, flags)
+  Action_builder.memo_build
+    (let open Memo.Build.O in
+    let libs = Result.ok_exn (Lib.DB.resolve_pps (Scope.libs scope) pps) in
```

Comment:
is it ok to add all these ok_ex here?

Reply:
I have been wondering about introducing a `raise : exn -> unit Memo.Build.t` function, and use that exclusively in the rules, for the same purpose